tests/raster/test_workingcopy.py

```diff
@@ -527,3 +528,110 @@ def _set_category_labels(pam_path, category_labels):
                     category.firstChild.nodeValue = category_labels[row_id]
 
         pam_path.write_text(parsed.toxml())
+
+
+def test_working_copy_edit__convert_to_cog(
+    cli_runner, data_archive, requires_gdal_info, requires_git_lfs, check_lfs_hashes
+):
+    with data_archive("raster/tif-aerial.tgz") as tif_aerial:
+        with data_archive("raster/aerial.tgz") as repo_path:
+            r = cli_runner.invoke(["diff", "--exit-code"])
+            assert r.exit_code == 0
+
+            shutil.copy(tif_aerial / "aerial.tif", repo_path / "aerial/aerial2.tif")
+
+            assert get_hash_and_size_of_file(repo_path / "aerial/aerial2.tif") == (
+                "bdbb58a399b60231f7a017fd76659efb0f5c1d82ab892248123d14d9a1e838e1",
+                393860,
+            )
+
+            r = cli_runner.invoke(["status"])
+            assert r.exit_code == 0, r.stderr
+            assert r.stdout.splitlines() == [
+                "On branch main",
+                "",
+                "Changes in working copy:",
+                '  (use "kart commit" to commit)',
+                '  (use "kart restore" to discard changes)',
+                "",
+                "  aerial:",
+                "    meta:",
+                "      1 updates",
+                "    tile:",
+                "      1 inserts",
+            ]
+
+            r = cli_runner.invoke(["diff"])
+            assert r.exit_code == 0, r.stderr
+            assert r.stdout.splitlines() == [
+                "--- aerial:meta:format.json",
+                "+++ aerial:meta:format.json",
+                "- {",
+                '-   "fileType": "geotiff",',
+                '-   "profile": "cloud-optimized"',
+                "- }",
+                "+ <<<<<<< ",
```

Comment:
I don't understand why there's a conflict here?

Reply:
This too may only make sense as part of a broader check that is not special cased.
Let's say your tiles have a band-schema of 32-bit RGBA.
If you insert more tiles with the 32-bit RGBA, you may commit.
If you delete all tiles and replace them with tiles with 24-bit RGB, you may also commit. This now counts as a meta-update - the schema.json meta item will change.
If you insert one tile that has 24-bit RGB, you may not commit. This shows up as a tile insert, and as a meta-item update - the old value is 32-bit RGBA, and the new value is a ListOfConflicts containing two values: (32-bit RGBA, 24-bit RGB).

format.json uses much same code as schema.json. So, right now you have a format.json saying geotiff, cloud-optimized.
If you insert more COGs, you may commit.
If you delete all tiles and replace with non-COG TIFFs, you may commit.
If you insert a single tile that is non-COG, you may not commit. This shows up as a tile insert, and as a meta-item update - the old value is geotiff+cloud-optimized, and the new value is a ListOfConflicts containing two values: (geotiff+cloud-optimized, and vanilla geotiff). That's what you can see here.

There was an idea at some point to have some kind of `--update-dataset-to-match-new-tiles` flag in addition to the opposite flag that we do have `--convert-to-dataset-format` but - we don't have one so far. (You could do this manually using `kart meta set`) So, if you get a conflict like this, where the new tile format doesn't match the old one, the best option is probably `--convert-to-dataset-format`. Maybe we should add a helpful hint to the diff letting the user know about this flag.

Reply:
yeah, that makes sense. However, I think we're going to hit this quite a bit:

* `kart import` _by default_ converts your files to COPC/COG, so most new tiled datasets will be COPC/COG
* Adding new tiles via committing _by default_ does not convert to the dataset format

Since users will inevitably use the same processing pipeline for the stuff they import and commit, I feel like almost all datasets that ever get updated will hit this conflict.

Should `--convert-to-dataset-format` just be the default?